lib/rust/prelude/src/debug.rs

```diff
@@ -126,3 +126,75 @@ impl Drop for TraceCopies {
         }
     }
 }
+
+
+
+// ====================
+// === LeakDetector ===
+// ====================
+
+/// A module containing an utility for detecting leaks.
```

Comment:
I was reading documentation of this module and the inner structs and I got the high-level overview, but there was no examples how this should be used and no explanation of how to use it. The constructor function documentation `Create enabled structure with appointed entity name (shared between all copies).` is not helpful either - I mean, it creates an "enabled" structure with a shared name, but then what? I think I'd need a little bit more docs here.

Also, I don't see how it's used in the new code, but that's probably it is somehow connected to the existing `TraceCopies` and I need some docs to understand it better.

Reply:
> Create enabled structure with appointed entity name (shared between all copies). is not helpful either - I mean, it creates an "enabled" structure with a shared name, but then what? I think I'd need a little bit more docs here.

Well, then it does exactly what struct's documentation says: 
* The underlying TraceCopies tracks copies, printing when they are created and dropped.
* Additionally, the existing copies are stored in [`TRACKED_OBJECTS`].
The documentation should be read with `TraceCopies` the `lead_detector::Trace` is a wrapper over it (and documentation says it).

What information do you still miss?

Reply:
Ok, I think adding a clear and explicit information that all the `Traces` are stored in `TRACKED_OBJECTS` when the leak detector is enabled is good enough, but I missed that from docs. Maybe someone else would find it obvious, but I don't. I'd be very thankful for adding a line with this explanation there.